### DISCUSSION TITLE
Rich entity history

### DISCUSSION THREAD

**Discussion by niemyjski** [home-assistant/frontend#7656] [Dashboard]
NOTE: This feature request may span into the core project.

I've very thankful that currently we have entity history and it works for the most part.
![image](https://user-images.githubusercontent.com/1020579/98826961-f17a0000-23fb-11eb-9ac5-1285b24fd2db.png)

But it really fails to provide value when you are trying to figure out why something actually happened as you have no context. And it makes it really hard to track it down to a specific automation or automation trigger. I'd really love if we could take this to the next level by being able to augment this information by providing additional data to service calls like a `source` and `reason`. I think it would look like this

![image](https://user-images.githubusercontent.com/1020579/98827454-83820880-23fc-11eb-90e6-7e98a8ac235c.png)

This would make it so you could easily see what triggered the light to turn on and this could work across automations and node red automations or any service call. You could even have rich interaction to go right to the automation to edit it.

I think this would also solve some issues where users are using a custom entity sensor to provide this information.

**Comment by KTibow** [comment]
Related links:
https://community.home-assistant.io/t/what-the-heck-turned-my-light-on/219488
https://github.com/home-assistant/frontend/pull/6682
https://github.com/home-assistant/core/pull/39194
